blocks/newsletter-modal/newsletter-modal.js

```diff
@@ -0,0 +1,101 @@
+import { getLibs } from '../../scripts/utils.js';
+
+function closeModal() {
+  const qModals = document.querySelectorAll('.dialog-modal');
+  if (qModals?.length) {
+    qModals.forEach((modal) => {
+      if (modal.nextElementSibling?.classList.contains('modal-curtain')) {
+        modal.nextElementSibling.remove();
+      }
+      modal.remove();
+    });
+    window.history.pushState('', document.title, `${window.location.pathname}${window.location.search}`);
+  }
+}
+
+function displayConfirmation(content, message, createTag) {
+  const confirmationText = createTag('p', { class: 'newsletter-modal-confirmation' }, message);
+  const confirmationClose = createTag('button', { class: 'newsletter-modal-confirmation-close' }, 'Close');
+
+  content.innerHTML = '';
+  content.append(confirmationText, confirmationClose);
+
+  confirmationClose.addEventListener('click', (e) => {
+    e.preventDefault();
+    closeModal();
+  });
+}
+
+export default async function init(block) {
+  const children = block.querySelectorAll(':scope > div');
+  if (children.length < 4) return;
+
+  block.innerHTML = '';
+  const { createTag } = await import(`${getLibs()}/utils/utils.js`);
+
+  const picture = children[0].querySelector('picture');
+  const text = children[1].querySelector('div > p:nth-child(1)').textContent;
+  const disclaimer = children[1].querySelector('div > p:nth-child(2)').textContent;
+
+  const emailText = children[2].querySelector('div p:nth-child(1)').textContent;
+  const emailPlaceholder = children[2].querySelector('div p:nth-child(2)').textContent;
+  const cta = children[2].querySelector('div p:nth-child(3)').textContent;
+
+  const successMsg = children[3].querySelector(':scope > div:nth-child(1) ').textContent;
+  const errorMsg = children[3].querySelector(':scope > div:nth-child(2) ').textContent;
+
+  const bannerContainer = createTag('div', { class: 'newsletter-modal-banner-container' }, picture);
+  const content = createTag('div', { class: 'newsletter-modal-content' });
```

Comment:
It seems somewhat wasteful to remove divs and then create new ones. On the other hand, it's potentially easier for devs to follow what's going on that way. I'd be curious to hear what other milo devs prefer.

Reply:
I tend to agree. 

If you can use what `plain.html` gives and add classes to the provided `divs`, to style correctly, that would be ideal and save some bytes.